### DISCUSSION TITLE
as.h  __afl_forkserver

### DISCUSSION THREAD

**Discussion by simple-somewhat** [AFLplusplus/AFLplusplus#1641] [General]
in afl-as.h main_payload_64 __afl_forkserver , the comment said "Enter the fork server mode to avoid the overhead of execve() calls. We\n"  "     push rdx (area ptr) twice to keep stack alignment neat.", I don’t understand，function call require stack aligned to a multiple of 16 bytes，because call will pushes 8 bytes onto stack，i think only push once is ok . sry for pool english, am i explain clear

**Comment by vanhauser-thc** [comment]
it works, it is outdated, nobody should use it anymore :)
only afl-gcc still uses this. but much better is afl-clang-fast which uses llvm plugins.
